lib/utils/compose.coffee

```diff
@@ -0,0 +1,746 @@
+		# here and bypass the logger.
+		prefix = chalk.blue('[Build]') + '   '
+
+		offset = 10 # account for escape sequences inserted for colouring
+		@_prefixWidth = offset + prefix.length + _.max(_.map(services, 'length'))
+		@_prefix = prefix
+
+		# these are to handle window wrapping
+		@_maxLineWidth = null
+		@_lineWidths = []
+
+		@_startTime = null
+		@_ended = false
+		@_cancelled = false
+		@_spinner = createSpinner()
+
+		@streams = streams
+
+	_handleEvent: (service, event) =>
+		@_serviceToDataMap[service] = event
+
+	_handleInterrupt: =>
+		@_cancelled = true
+		@end()
+		process.exit(130) # 128 + SIGINT
+
+	start: =>
+		process.on('SIGINT', @_handleInterrupt)
+		@_tty.hideCursor()
+		@_services.forEach (service) =>
+			@streams[service].write({ status: 'Preparing...' })
+		@_runloop = createRunLoop(@_display)
+		@_startTime = Date.now()
+
+	end: (summary = null) =>
+		return if @_ended
+		@_ended = true
+		process.removeListener('SIGINT', @_handleInterrupt)
+		@_runloop.end()
+		@_runloop = null
+
+		@_clear()
+		@_renderStatus(true)
+		@_renderSummary(summary ? @_getServiceSummary())
+		@_tty.showCursor()
+
+	_display: =>
+		@_clear()
+		@_renderStatus()
+		@_renderSummary(@_getServiceSummary())
+		@_tty.cursorUp(@_services.length + 1) # for status line
+
+	_clear: ->
+		@_tty.deleteToEnd()
+		@_maxLineWidth = @_tty.currentWindowSize().width
+
+	_getServiceSummary: ->
+		_ = require('lodash')
+
+		services = @_services
+		serviceToDataMap = @_serviceToDataMap
+
+		_(services).map (service) ->
+			{ status, progress, error } = serviceToDataMap[service] ? {}
+			if error
+				return "#{error}"
+			else if progress
+				bar = renderProgressBar(progress, 20)
+				return "#{bar} #{status}" if status
+				return "#{bar}"
+			else if status
+				return "#{status}"
+			else
+				return 'Waiting...'
+		.map (data, index) ->
+			[ services[index], data ]
+		.fromPairs()
+		.value()
+
+	_renderStatus: (end = false) ->
+		moment = require('moment')
+		require('moment-duration-format')(moment)
+
+		@_tty.clearLine()
+		@_tty.write(@_prefix)
+		if end and @_cancelled
+			@_tty.writeLine('Build cancelled')
+		else if end
+			serviceCount = @_services.length
+			serviceStr = if serviceCount is 1 then '1 service' else "#{serviceCount} services"
+			runTime = Date.now() - @_startTime
+			durationStr = moment.duration(runTime // 1000, 'seconds').format()
+			@_tty.writeLine("Built #{serviceStr} in #{durationStr}")
+		else
+			@_tty.writeLine("Building services... #{@_spinner()}")
+
+	_renderSummary: (serviceToStrMap) ->
+		_ = require('lodash')
+		chalk = require('chalk')
+		truncate = require('cli-truncate')
+		strlen = require('string-width')
+
+		@_services.forEach (service, index) =>
+			str = _.padEnd(@_prefix + chalk.bold(service), @_prefixWidth)
+			str += serviceToStrMap[service]
+			if @_maxLineWidth?
+				str = truncate(str, @_maxLineWidth)
+			@_lineWidths[index] = strlen(str)
+
+			@_tty.clearLine()
+			@_tty.writeLine(str)
```

Comment:
It would be much nicer to get the table here that the builder uses, I'll even seperate it out into a seperate module for you if you want.

Reply:
That would be nice, but it'll have to come in a subsequent PR. The command already presents a summary of useful info and it's good enough for now.